client-assertion-validation/src/test/scala/it/pagopa/interop/clientassertionvalidation/ClientAssertionValidationSpec.scala

```diff
@@ -91,6 +91,14 @@ class ClientAssertionValidationSpec extends AnyWordSpecLike {
       ) shouldBe Left(NonEmptyList.one(InvalidPurposeIdFormat("not-an-uuid")))
     }
 
+    "fail when purpose id is ampty string" in {
+      val assertion = fastClientAssertionJWT(customClaims = Map(PURPOSE_ID_CLAIM -> ""))
+
+      validateClientAssertion(Some(clientId.toString), assertion, clientAssertionType, grantType)(
+        jwtValidator
+      ) shouldBe Left(NonEmptyList.one(InvalidPurposeIdFormat("")))
```

Comment:
Testing what I suggested a few lines before
```suggestion
      ) shouldBe Left(NonEmptyList.one(InvalidPurposeIdFormat("   ")))
```